File: src/TestArrays.java
public class TestArrays
{
    private static final int MINUSFIVE = -5;
    private static final int TEN = 10;

    private static enum greek {ALPHA, BETA, GAMMA};
    private static enum days {MON, TUE, WED, THU, FRI, SAT, SUN};

    private static int a1a[] = new int[10], a1b[] = new int[10];
    private static double a2[] = new double[21];
    private static boolean a3[] = new boolean[4];
    private static greek a4[] = new greek[3];
    private static days a5[][] = new days[21][16];
    private static days a6[][][] = new days[21][16][8];
    private static char chs[] = new char[32];
    private static int ua1[] = new int[26];
    private static int ua2[][] = new int[21][5];
    private static greek ua3[][][] = new greek[2][21][3];

    public static void main(String[] args)
    {
        System.out.println("Done!");
    }
}

File: TestArrays.pas
PROGRAM TestArrays;

CONST
    MINUSFIVE = -5;
    TEN = 10;

TYPE
    greek = (ALPHA, BETA, GAMMA);
    days = (MON, TUE, WED, THU, FRI, SAT, SUN);
    weekdays = MON..FRI;
    sr1 = 5..25;
    sr2 = MINUSFIVE..TEN;
    sr3 = -TEN..-3;
    
    ar1 = ARRAY [1..TEN] OF integer;
    ar2 = ARRAY [sr1] OF real;
    ar3 = ARRAY [WED..SAT] of boolean;
    ar4 = ARRAY [(SMALL, MEDIUM, LARGE)] OF greek;
    char32 = ARRAY [1..32] OF char;                  
    
    ar5 = ARRAY [sr1] OF ARRAY [sr2] OF weekdays;
    ar6 = ARRAY [sr1, sr2, sr3] OF weekdays;

VAR
    a1a, a1b : ar1;
    a2 : ar2;
    a3 : ar3;
    a4 : ar4;
    a5 : ar5;
    a6 : ar6;
    chs : char32;

    ua1 : ARRAY ['a'..'z'] OF (JAN, FEB, MAR, APR);
    ua2 : ARRAY [sr1] OF ARRAY [8..12] OF JAN..APR;
    ua3 : ARRAY [boolean, sr1, (RED, WHITE, BLUE)] 
                                         OF greek;
                
BEGIN
END.
